adafruit_io/adafruit_io.py

```diff
@@ -0,0 +1,257 @@
+# The MIT License (MIT)
+#
+# Copyright (c) 2019 Brent Rubell for Adafruit Industries
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+"""
+`adafruit_io`
+================================================================================
+
+A CircuitPython/Python library for communicating with Adafruit IO over WiFi
+
+* Author(s): Brent Rubell for Adafruit Industries
+
+Implementation Notes
+--------------------
+
+**Software and Dependencies:**
+
+* Adafruit CircuitPython firmware for the supported boards:
+    https://github.com/adafruit/circuitpython/releases
+
+* Adafruit's ESP32SPI library:
+    https://github.com/adafruit/Adafruit_CircuitPython_ESP32SPI
+"""
+from adafruit_io.adafruit_io_errors import AdafruitIO_RequestError, AdafruitIO_ThrottleError
+
+__version__ = "0.0.0-auto.0"
+__repo__ = "https://github.com/adafruit/Adafruit_CircuitPython_Adafruit_IO.git"
+
+class RESTClient():
+    """
+    REST Client for interacting with the Adafruit IO API.
+    """
+    def __init__(self, adafruit_io_username, adafruit_io_key, wifi_manager):
+        """
+        Creates an instance of the Adafruit IO REST Client.
+        :param str adafruit_io_username: Adafruit IO Username
+        :param str adafruit_io_key: Adafruit IO Key
+        :param wifi_manager: WiFiManager object from adafruit_esp32spi_wifimanager
+        """
+        self.username = adafruit_io_username
+        self.key = adafruit_io_key
+        if wifi_manager:
```

Comment:
I don't understand what condition this is checking...
- `wifi_manager` is a positional argument in `__init__`, so if its missing the class instantiation will fail.
- Only checking that `wifi_manager` is a non-False object, allows `wifi_manager` to be anything. `True` would even suffice, which allows `self.wifi == True`.

I would suggest using `ininstance()` or `type()` to check that a valid wifi_manager has been supplied. I assume that the type is `adafruit_esp32spi_wifimanager.WifiManager`, but I haven't used these libraries yet.